cloud/aws/README.md

```diff
@@ -1,10 +1,116 @@
-# VAST cloud deployment
+# VAST AWS deployment
 
-Deploy VAST easily to AWS with Terraform.
+*Disclaimer: the native cloud architecture is still highly experimental and
+subject to change without notice. Do not use in production yet.*
 
-## Proposed architecture
+Running VAST on AWS relies on two serverless building blocks:
 
-### Infrastructure overview
+1. [Fargate](https://aws.amazon.com/fargate/) for VAST server nodes
+2. [Lambda](https://aws.amazon.com/lambda/) as compute service to perform
+   tasks, such as executing queries and ingesting data ad hoc
+
+For storage on the VAST server in Fargate, VAST uses
+[EFS](https://aws.amazon.com/efs/) as a drop-in network filesystem with a
+reasonable throughput-latency trade-off. We are still evaluating local
+storage optimization using [EBS](https://aws.amazon.com/ebs/) and
+high-throughput scale-out using [S3](https://aws.amazon.com/s3/).
+
+### Usage
+
+The provided [Makefile](Makefile) wraps [Terraform](https://www.terraform.io/)
+and `aws` CLI commands for a streamlined UX. You need the following tools
+installed locally:
+
+- Terraform (> v1): `terraform`
+- AWS CLI (v2): `aws`
+- Docker: `docker`
+- `jq` for parsing `aws` results
+
+To avoid having to provide required Terraform variables manually, the Makefile
+looks for a file `default.env` in the current directory. It must define the
+following variables:
+
+- `aws_region`: the region where to deploy the VPC
+
+- `vpc_id`: an existing VPC to which you plan to attach your VAST stack. You
+  can use `aws ec2 describe-vpcs --region $region` to list available VPCs.
+
+- `subnet_cidr`: the subnet *within* the VPC where the VAST stack will be
+  placed. Terraform will create a this subnet and it must not overlap with an
+  existing subnet in this VPC and VPCs peered to it. You can use
+  `aws ec2 describe-subnets --region $region` to list existing subnets to pick
+  a non-overlapping one within the VPC subnet.
+
+Here's an example:
+
+```bash
+vpc_id=vpc-059a7ec8aac174fc9
+subnet_cidr=172.31.48.0/24
+aws_region=eu-north-1
+```
+
+Next, we take a look at some use cases. To setup Fargate and one Lambda
+instance, use `make deploy`. You can tear down the resources using the dual
+command, `make destroy`. See `make help` for a brief description of available
+commands.
+
+Caveats:
+
+- Access to the VAST server is enforced by limiting inbound traffic to its
+  local private subnet.
+
+- A NAT Gateway is created automatically, you cannot specify an existing one
```

Comment:
```suggestion
- A NAT Gateway is created automatically, you cannot specify an existing one. It will be billed at [an hourly rate](https://aws.amazon.com/vpc/pricing/) even when you aren't running any workload, until you tear down the entire stack.
```